volatility/framework/plugins/mac/check_syscall.py

```diff
@@ -23,13 +24,16 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
             requirements.TranslationLayerRequirement(name = 'primary',
                                                      description = 'Memory layer for the kernel',
                                                      architectures = ["Intel32", "Intel64"]),
-            requirements.SymbolTableRequirement(name = "darwin", description = "Mac kernel symbols")
+            requirements.SymbolTableRequirement(name = "darwin", description = "Mac kernel symbols"),
+            requirements.PluginRequirement(name = 'lsmod', plugin = lsmod.Lsmod, version = (1, 0, 0))
         ]
 
-    def _generator(self):
+    def _generator(self, mods: Iterator[Any]):
```

Comment:
Since this is not a class method, it's ok to use `self.config` to look up things like `primary` and `darwin`.  Also, because it's private and can only be used from this class, it seems a bit strange to have some functionality in the run method, and the rest in the generator?  I'd move the mods creating code into the top of the generator and then everything's easy to read in one go (unless you're intending to use _generator from within this class more than once?).  Not a show stopper, but seems a bit weird given the method can't ever be called by anyone else...